saleor/graphql/checkout/mutations.py

```diff
@@ -1032,6 +1046,139 @@ def perform_mutation(
+                    )
+                }
+            )
+
+        checkout.shipping_method = shipping_method
+        checkout.collection_point = None
+        checkout.save(
+            update_fields=["shipping_method", "collection_point", "last_change"]
+        )
+        recalculate_checkout_discount(
+            manager, checkout_info, lines, info.context.discounts
+        )
+        manager.checkout_updated(checkout)
+        return CheckoutDeliveryMethodUpdate(checkout=checkout)
+
+    @classmethod
+    def perform_on_collection_point(
+        cls, info, collection_point_id, checkout_info, lines, checkout, manager
+    ):
+        collection_point = cls.get_node_or_error(
+            info,
+            collection_point_id,
+            only_type=Warehouse,
+            field="collection_point_id",
+            qs=warehouse_models.Warehouse.objects.select_related("address"),
+        )
+        collection_point_is_valid = clean_delivery_method(
+            checkout_info=checkout_info, lines=lines, method=collection_point
+        )
+        if not collection_point_is_valid:
+            raise ValidationError(
+                {
+                    "delivery_method": ValidationError(
+                        "This pick up point is not applicable.",
+                        code=CheckoutErrorCode.SHIPPING_METHOD_NOT_APPLICABLE,
+                    )
+                }
+            )
+        checkout.collection_point = collection_point
+        checkout.shipping_method = None
+        checkout.save(
+            update_fields=["collection_point", "shipping_method", "last_change"]
+        )
+        manager.checkout_updated(checkout)
+        return CheckoutDeliveryMethodUpdate(checkout=checkout)
+
+    @classmethod
+    def perform_mutation(
+        cls,
+        _,
+        info,
+        token=None,
+        shipping_method_id=None,
+        collection_point_id=None,
+    ):
+
+        checkout = get_checkout_by_token(token)
+
+        manager = info.context.plugins
+        lines = fetch_checkout_lines(checkout)
+        checkout_info = fetch_checkout_info(
+            checkout, lines, info.context.discounts, manager
+        )
+        if not is_shipping_required(lines):
+            raise ValidationError(
+                {
+                    "delivery_method": ValidationError(
+                        ERROR_DOES_NOT_SHIP,
+                        code=CheckoutErrorCode.SHIPPING_NOT_REQUIRED,
+                    )
+                }
+            )
+        if shipping_method_id is not None and collection_point_id is not None:
+            raise ValidationError(
+                {
+                    "delivery_method": ValidationError(
+                        "Only one delivery method should be picked - C&C or Shipping",
+                        code=CheckoutErrorCode.INVALID,
+                    )
+                }
+            )
+        return (
```

Comment:
For better readability, can we just call it inside normal `if` condition ?  It won't add a lot of additional lines but it will be easier to read.

Reply:
👍  Done.